### PR TITLE
Initial Project Setup

### LINKED ISSUE #1: Setup initial code with linting and pre-commit hooks
This task sets up the initial codebase. The following are the tasks that will be involved:
```[tasklist]
### Tasks
- [x] Setup a basic FastAPI application with one endpoint that simply returns "Hello World".
- [x] Setup a virtual environment.
- [x] Setup .gitignore file (https://github.com/github/gitignore/blob/main/Python.gitignore)
- [x] Add a requirements.txt file.
- [x] Add a basic passing test.
- [x] Add a pre-commit hook that runs flake8 and also runs the tests.
```

### PR DESCRIPTION
**Pull request by ConradKash**
I am done adding linting and some tests.
Please try out on your local

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Environment
+venv/
```
Comment (ignatius-k) [inline .gitignore]: @ConradKash could you use the other link to python_.gitignore file @cmplx-xyttmt shared
cause ___pycache___ folderz are still being pushed

**Review by Ignatius-K — CHANGES_REQUESTED** [review]
Some changes in the .gitignore are needed

File: requirements.txt
Comment (cmplx-xyttmt) [inline requirements.txt:1]: I don't see flake8 in this list of dependencies.

**Review by cmplx-xyttmt — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (ConradKash) [reply .gitignore]: Yeah you are right. I had forgotten thank you.

File: requirements.txt
Comment (ConradKash) [reply requirements.txt:1]: My bad. i was running it as a pre-commit hook

Comment (ConradKash) [reply requirements.txt:1]: I have followed up the gitignore and tried watching videos even by the __pycache__ folder doesnt seem to go

Comment (ConradKash) [reply requirements.txt:1]: Please support. Its still getting commmited

**Review by Ignatius-K — APPROVED** [review]
Everything seems good

File: requirements.txt
Comment (cmplx-xyttmt) [reply requirements.txt:1]: that's because you had already committed them. Just leave them for nwo

**Pull request merged**